### ISSUE TITLE
`AutocorrectLogic#safe_autocorrect?` may be returning true erroneously

### ISSUE THREAD

**Issue by searls** [rubocop/rubocop#11715]
It appears like `safe_autocorrect?` may be returning `true` for cases where autocorrection is not safe.

For example, [Lint/BooleanSymbol](https://docs.rubocop.org/rubocop/cops_lint.html#lintbooleansymbol) supports autocorrection, but the docs report that it is unsafe. Additionally, the RuboCop's [default.yml agrees](https://github.com/rubocop/rubocop/blob/master/config/default.yml#L1596) and has `SafeAutoCorrect: false`. 

However, if you were to instantiate the cop and ask if it was safe, it would return true:

```ruby
RuboCop::Cop::Lint::BooleanSymbol.new.safe_autocorrect?
=> true
```

Is this a bug? It's affecting which cops Standard thinks are safe to autocorrect, so I'm trying to find the proper way to ascertain whether a violation is safely autocorrectable programmatically. 

Thank you for all your hard work!!! 💚

**Comment by searls** [comment]
@koic, could you take a look at this? I am concerned Standard may be running unsafe autocorrections as a result 😬

**Comment by koic** [comment]
Yeah, I'll look into it later. Thank you for the feedback!

**Comment by koic** [comment]
This looks like unexpected behavior. Either way I think it's a bug and opened #11729 as a patch.

**Comment by searls** [comment]
Thanks!!
